0000-navigator-view-api.md

```diff
@@ -0,0 +1,89 @@
+* Start Date: 2018-01-29
+* RFC PR: (leave this empty)
+* React Navigation Issue: (leave this empty)
+
+# Summary
+
+Introduce a simpler API for createNavigator and navigation views. By providing "sceneDescriptors" to the navigation view, it does not need to compute options or manually create the child navigation prop.
+
+Down the line, this will enable us to deprecate `addNavigationHelpers`, which is an avoidable burden for redux apps and when implementing custom navigators. Eventually, each navigator may provide a different set of helpers.
+
+# Basic example
+
+Using a new `createNavigator` API and `screenDescriptors` prop, this is simplest possible custom navigation view:
+
+```js
+const CustomNavigationView = ({navigation, screenDescriptors}) => {
+  const {routes, index} = navigation;
+  const descriptor = screenDescriptors[routes[index].key];
+  const ScreenComponent = descriptor.getComponent();
+  return (
+    <ScreenComponent
+      navigation={descriptor.navigation}
+      options={descriptor.options}
+    />
+  );
+};
+const router = StackRouter({...}, {...});
+const CustomNavigator = createNavigator(CustomNavigationView, router);
+```
+
+This `<CustomNavigator />` will have behave like a stack, only rendering the active scene. It can present screens as defined by the `router`.
+
+# Motivation
+
+This change prevents the router from being passed directly into the navigation view, which was causing coupling between the concepts. This change will allow for faster innovation on the navigation views, and allow them to be consumed by other navigation libraries.
+
+This also makes it easier to implement navigation views, because the children navigation props have already been provided within `props.sceneDescriptors`
+
+# Detailed design
+
+Navigators will now be created via a simpler API:
+
+`const Navigator = createNavigator(NavigationView, router, navigationConfig);`
+
+In this example, navigationConfig is an object that will be passed as a prop to the navigtion view. The `NavigationView` takes the following props:
+
+* `navigation` - Navigation prop with `dispatch`, `state`, and `addListener`
```

Comment:
also with `navigate`, at least for now?

Reply:
Sure, I'll explain that we currently have all of the helpers on this, including navigate